test/server/guarddog_impl_test.cc

```diff
@@ -24,21 +24,22 @@ namespace Server {
 class GuardDogDeathTest : public testing::Test {
 protected:
   GuardDogDeathTest()
-      : config_kill_(1000, 1000, 100, 1000), config_multikill_(1000, 1000, 1000, 500),
-        time_point_(std::chrono::system_clock::now()) {}
+      : config_kill_(1000, 1000, 100, 1000), config_multikill_(1000, 1000, 1000, 500) {}
 
   /**
    * This does everything but the final forceCheckForTest() that should cause
    * death for the single kill case.
    */
   void SetupForDeath() {
     InSequence s;
-    EXPECT_CALL(time_source_, currentSystemTime()).WillRepeatedly(testing::Return(time_point_));
+    EXPECT_CALL(time_source_, currentSystemTime())
```

Comment:
nit: EXPECT_CALL() with WillRepeatedly is an anti-pattern in that it doesn't actually expect anything since 0 times will pass. I would make this ON_CALL(...).WillByDefault(...). (Same for all below).